packages/dds/map/src/directory.ts

```diff
@@ -1132,22 +1209,29 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 		}
 
 		// Create the sub directory locally first.
-		const isNew = this.createSubDirectoryCore(subdirName, true);
-
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		const subDir: IDirectory = this._subdirectories.get(subdirName)!;
+		const isNew = this.createSubDirectoryCore(
+			subdirName,
+			true,
+			-1,
+			this.runtime.clientId ?? "detached",
+		);
+		const subDir = this._subdirectories.get(subdirName);
+		assert(subDir !== undefined, "subdirectory should exist after creation");
 
 		// If we are not attached, don't submit the op.
 		if (!this.directory.isAttached()) {
 			return subDir;
 		}
 
-		const op: IDirectoryCreateSubDirectoryOperation = {
-			path: this.absolutePath,
-			subdirName,
-			type: "createSubDirectory",
-		};
-		this.submitCreateSubDirectoryMessage(op, !isNew);
+		// Only submit the op, if it is newly created.
+		if (isNew) {
+			const op: IDirectoryCreateSubDirectoryOperation = {
+				path: this.absolutePath,
+				subdirName,
+				type: "createSubDirectory",
+			};
+			this.submitCreateSubDirectoryMessage(op, !isNew);
```

Comment:
The second parameter to submitCreateSubDirectoryMessage can go away since your change will make it always false. (and remove it from local op metadata)

Reply:
done